src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/ExcelLoader.php

```diff
@@ -0,0 +1,253 @@
+    {
+        $this->dateFormat = $format;
+
+        return $this;
+    }
+
+    public function withDateTimeFormat(string $format) : self
+    {
+        $this->dateTimeFormat = $format;
+
+        return $this;
+    }
+
+    public function withHeader(bool $withHeader = true) : self
+    {
+        $this->withHeader = $withHeader;
+
+        return $this;
+    }
+
+    public function withHeaderStyle(Style $style) : self
+    {
+        $this->headerStyle = $style;
+
+        return $this;
+    }
+
+    public function withSheetName(?string $sheetName) : self
+    {
+        if ($sheetName !== null && $this->sheetNameEntryName !== null) {
+            throw new InvalidArgumentException('Cannot set both sheetName and sheetNameFromEntry. These options are mutually exclusive.');
+        }
+
+        if ($sheetName !== null) {
+            SheetNameAssertion::assert($sheetName);
+        }
+
+        $this->sheetName = $sheetName;
+
+        return $this;
+    }
+
+    public function withSheetNameFromEntry(string $entryName) : self
+    {
+        if ($this->sheetName !== null) {
+            throw new InvalidArgumentException('Cannot set both sheetName and sheetNameFromEntry. These options are mutually exclusive.');
+        }
+
+        $this->sheetNameEntryName = $entryName;
+
+        return $this;
+    }
+
+    public function withTimeFormat(string $format) : self
+    {
+        $this->timeFormat = $format;
+
+        return $this;
+    }
+
+    public function withWriter(ExcelWriter $writer) : self
+    {
+        $this->writerType = $writer;
+
+        return $this;
+    }
+
+    public function withWriterOptions(OdsOptions|XlsxOptions $options) : self
+    {
+        $this->writerOptions = $options;
+
+        return $this;
+    }
+
+    private function getWorkbookManager() : WorkbookManager
+    {
+        if ($this->workbookManager === null) {
+            $this->workbookManager = new WorkbookManager(
+                writerType: $this->resolveWriterType(),
+                options: $this->writerOptions,
+            );
+        }
+
+        return $this->workbookManager;
+    }
+
+    /**
+     * @return null|array<int, null|Style>
+     */
+    private function resolveCellStyles(Row $row, int $rowIndex, string $sheetName) : ?array
+    {
+        if ($this->cellStyler === null) {
+            return null;
+        }
+
+        $styles = [];
+        $columnIndex = 0;
+
+        foreach ($row->entries() as $entry) {
+            $styles[$columnIndex] = $this->cellStyler->style($entry, $rowIndex + 1, $columnIndex, $sheetName);
+            $columnIndex++;
+        }
+
+        return $styles;
+    }
+
+    private function resolveSheetName(Row $row) : string
+    {
+        if ($this->sheetNameEntryName !== null && $row->has($this->sheetNameEntryName)) {
+            $value = $row->get($this->sheetNameEntryName)->value();
+
+            if (\is_string($value) && $value !== '') {
```

Comment:
I suppose we can omit that first check, as we use strict comparison anyway, correct?
```suggestion
            if ($value !== '') {
```

Reply:
Note: after playing with real code, this is correct.